src/libraries/Microsoft.NETCore.Platforms/tests/GenerateRuntimeGraphTests.cs

```diff
@@ -21,38 +23,219 @@ public GenerateRuntimeGraphTests(ITestOutputHelper output)
             _engine = new TestBuildEngine(_log);
         }
 
-        [Fact]
-        public void CanCreateRuntimeGraph()
-        {
-            string runtimeFile = "runtime.json";
+        const string DefaultRuntimeFile = "runtime.json";
+        private static ITaskItem[] DefaultRuntimeGroupItems { get; } = GetDefaultRuntimeGroupItems();
 
+        private static ITaskItem[] GetDefaultRuntimeGroupItems()
+        {
             Project runtimeGroupProps = new Project("runtimeGroups.props");
 
             ITaskItem[] runtimeGroups = runtimeGroupProps.GetItems("RuntimeGroupWithQualifiers")
                                                  .Select(i => CreateItem(i)).ToArray();
 
             Assert.NotEmpty(runtimeGroups);
 
+            return runtimeGroups;
+        }
+
+        private static ITaskItem CreateItem(ProjectItem projectItem)
+        {
+            TaskItem item = new TaskItem(projectItem.EvaluatedInclude);
+            foreach (var metadatum in projectItem.Metadata)
+            {
+                item.SetMetadata(metadatum.Name, metadatum.EvaluatedValue);
+            }
+            return item;
+        }
+
+        [Fact]
+        public void CanCreateRuntimeGraph()
+        {
             // will generate and compare to existing file.
             GenerateRuntimeGraph task = new GenerateRuntimeGraph()
             {
                 BuildEngine = _engine,
-                RuntimeGroups = runtimeGroups,
-                RuntimeJson = runtimeFile
+                RuntimeGroups = DefaultRuntimeGroupItems,
+                RuntimeJson = DefaultRuntimeFile,
+                UpdateRuntimeFiles = false
             };
             task.Execute();
 
             _log.AssertNoErrorsOrWarnings();
         }
 
-        private static ITaskItem CreateItem(ProjectItem projectItem)
+
+        [Fact]
+        public void CanIgnoreExistingInferRids()
         {
-            TaskItem item = new TaskItem(projectItem.EvaluatedInclude);
-            foreach(var metadatum in projectItem.Metadata)
+            // will generate and compare to existing file.
```

Comment:
Should we assert that the generated file has the additional rids?

Reply:
Ah, this test adds rids that already exist?

Reply:
Right. This test passes if the addition of RIDs make no change to the checked in runtime.json